modelling/src/main/java/org/axonframework/modelling/command/AggregateAnnotationCommandHandler.java

```diff
@@ -395,7 +395,7 @@ public Object handle(CommandMessage<?> command) throws Exception {
             VersionedAggregateIdentifier iv = commandTargetResolver.resolveTarget(command);
             Aggregate<T> inst = repository.loadOrCreate(iv.getIdentifier(), factoryMethod);
             Object result = inst.handle(command);
-            assertThat(inst.identifier(), id -> id != null && id.equals(iv.getIdentifier()),
+            assertThat(inst.identifier(), id -> id != null && String.valueOf(id).equals(iv.getIdentifier()),
```

Comment:
I thought I already fixed this in [this](https://github.com/AxonFramework/AxonFramework/pull/1355/commits/d4b5ef9709ea0e2626d45bbfd0aace4db3c99825) commit.